pkg/traffic/gateway.go

```diff
@@ -10,19 +10,38 @@ import (
 	gatewayv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
 
 	"github.com/Kuadrant/multicluster-gateway-controller/pkg/_internal/slice"
+	"github.com/Kuadrant/multicluster-gateway-controller/pkg/controllers/gateway"
 )
 
 type GatewayInterface interface {
 	Interface
 	GetListenerByHost(host string) *gatewayv1beta1.Listener
 }
 
-func NewGateway(g *gatewayv1beta1.Gateway) GatewayInterface {
-	return &Gateway{Gateway: g}
+func NewGateway(g *gatewayv1beta1.Gateway) (*Gateway, error) {
+	gw := &Gateway{Gateway: g, isMultiCluster: false}
+
+	for i, address := range gw.Status.Addresses {
+		if i == 0 {
+			gw.isMultiCluster = isMultiClusterAddressType(*address.Type)
+			continue
+		}
+		if gw.isMultiCluster == isMultiClusterAddressType(*address.Type) {
+			continue
+		}
+		return nil, fmt.Errorf("gateway is invalid: inconsistent status addresses")
+
+	}
+	return gw, nil
+}
+
+func isMultiClusterAddressType(addressType gatewayv1beta1.AddressType) bool {
+	return addressType == gateway.MultiClusterIPAddressType || addressType == gateway.MultiClusterHostnameAddressType
 }
 
 type Gateway struct {
```

Comment:
This is already here, but we probably should avoid calling it "Gateway", GatewayWrapper would probably be better similar to what the kuadrant operator does see https://github.com/Kuadrant/kuadrant-operator/blob/main/pkg/common/gatewayapi_utils.go#L355

Reply:
`GatewayWrapper` it is. Renamed.